### ISSUE TITLE
global: reduce pip backtracking for Python 3.10

### ISSUE THREAD

**Issue by GraemeWatt** [HEPData/hepdata#669]
Trying to install the HEPData software in a Python 3.10 environment gives an info message:

```
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.
```

followed eventually by an error message:
```
ERROR: Exception:
pip._vendor.resolvelib.resolvers.ResolutionTooDeep: 200000
```
We might need to simplify the dependency resolution, for example, by combining the dependencies from `requirements.txt` and `setup.py`, to allow use of Python 3.10.  [End-of-life](https://endoflife.date/python) for Python 3.9 is not until 5th October 2025, so the move to Python 3.10 is not urgent.  See https://pip.pypa.io/warnings/backtracking for more information.

**Comment by GraemeWatt** [comment]
I just checked whether this issue still applies and it seems to be resolved with the latest `main` branch.  I've opened a new issue #851 to upgrade from Python 3.9 and I'll close this issue for now.
